packages/babel-core/src/tools/build-external-helpers.ts

```diff
@@ -94,7 +95,10 @@ function buildModule(allowlist?: Array<string>) {
     exportNamedDeclaration(
       null,
       Object.keys(refs).map(name => {
-        return exportSpecifier(cloneNode(refs[name]), identifier(name));
+        return exportSpecifier(
+          cloneNode(refs[name] as t.Identifier),
```

Comment:
We can add function overload to buildHelpers:

```ts
function buildHelpers(
  body: t.Statement[],
  namespace: t.Expression,
  allowlist?: Array<string>,
): Record<string, t.MemberExpression>;
function buildHelpers(
  body: t.Statement[],
  namespace: null,
  allowlist?: Array<string>,
): Record<string, t.Identifier>
```

so TS will know that `refs[name]` must be an identifier when `namespace` is `null`.